extensions/librdkafka/PublishKafka.cpp

```diff
@@ -411,6 +414,21 @@ bool PublishKafka::createNewTopic(const std::shared_ptr<KafkaConnection> &conn,
 
   value = "";
   if (context->getProperty(DeliveryGuarantee.getName(), value) && !value.empty()) {
+    /*
+     * Because of a previous error in this processor, the default value of this property was "DELIVERY_ONE_NODE".
+     * As this is not a valid value for "request.required.acks", the following rd_kafka_topic_conf_set call failed,
+     * but because of an another error, this failure was silently ignored, meaning that the the default value for
+     * "request.required.acks" did not change, and thus remained "-1". This means that having "DELIVERY_ONE_NODE" as
+     * the value of this property actually caused the processor to wait for delivery ACKs from ALL nodes, instead
+     * of just one. In order not to break configurations generated with earlier versions and keep the same behaviour
+     * as they had, we have to map "DELIVERY_ONE_NODE" to "-1" here.
+     */
+    if (value == "DELIVERY_ONE_NODE") {
+      value = "-1";
+      logger_->log_warn("Using DELIVERY_ONE_NODE as the Delivery Guarantee property is deprecated and is translated to -1 "
```

Comment:
I think we should have a follow-up ticket to address this is 1.0

Reply:
Agreed: https://issues.apache.org/jira/browse/MINIFICPP-1051